Assets/Scripts/Combat/Abilities/FireBall.cs

```diff
@@ -0,0 +1,278 @@
+            float fireballDamage = (M - m) / (Mathf.PI / 2) * Mathf.Atan(s / d) + m;
+
+            return fireballDamage;
+        }
+
+        protected override void EndAbility()
+        {
+            Debug.Log($"Fireball Damage total: {CurrentDamage}");
+
+            // Deal damage to defender, wait
+            GameObject victim = TargetedCombatants[Random.Range(0, TargetedCombatants.Length)];
+            
+            StartCoroutine(LaunchFireball(victim));
+
+            CurrentFireballPhase = FireballPhase.Inactive;
+        }
+
+        private IEnumerator LaunchFireball(GameObject target)
+        {
+            float launchDuration = 1f;
+            float elapsedTime = 0;
+            Vector3 origin = Fireball.transform.position;
+
+            while (elapsedTime < launchDuration)
+            {
+                Fireball.transform.position = Vector3.Lerp(
+                    origin, 
+                    target.transform.position,
+                    elapsedTime / launchDuration);
+                elapsedTime += Time.deltaTime;
+                yield return null;
+            }
+
+            Fireball.SetActive(false);
+
+            Attack attack = new Attack(CurrentDamage);
+            target.GetComponent<Combatant>()
+                    .Defend(attack);
+
+            CombatSystem.EndTurn(this.GetComponentInParent<Combatant>().gameObject);
+        }
+
+        private void GrowFireball()
+        {
+            Debug.Log("GrowFireball");
+            var currentKey = ExpectedDirections[0];
+            ExpectedDirections.Remove(currentKey);
+            ExpectedDirections.Add(currentKey);
+
+            FireballSize += FireballGrowth;
+            FireballScalingElapsedTime = 0;
+        }
+
+        private void ShrinkFireball()
+        {
+            Debug.Log("ShrinkFireball");
+
+            FireballSize -= CurrentFireballCyclePhase == FireballCyclePhase.Unstable ? FireballShrinkUnstable : FireballShrinkNormal;
+            if (FireballSize < 0)
+                FireballSize = 0f;
+            FireballScalingElapsedTime = 0;
+        }
+
+        private bool IsFireballKeyDown()
+        {
+            return Input.GetKeyDown(KeyCode.W) ||
+                Input.GetKeyDown(KeyCode.D) ||
+                Input.GetKeyDown(KeyCode.S) ||
+                Input.GetKeyDown(KeyCode.A);
+        }
+
+        public new void StartAbility()
+        {
+            Debug.Log("Starting fireball ability");
+
+            CurrentDamage = 0;
+            FireballSize = 1f;
+            CurrentFireballCycle = 0;
+
+            base.StartAbility();
+        }
+        protected override void ContinueAbilityAfterTargeting()
+        {
+            StartFireballPhase();
+        }
+
+        private void StartFireballPhase()
+        {
+            this.CurrentFireballPhase = FireballPhase.Growth;
+
+            Fireball.SetActive(true);
+            Fireball.transform.position = transform.position + 5f * Vector3.up;
```

Comment:
What is the 5f here for?

Reply:
Building the fireball overhead the player. Gonna extract it now!